packages/core-flows/src/definition/cart/workflows/complete-cart.ts

```diff
@@ -0,0 +1,19 @@
+import { CompleteCartWorkflowInputDTO } from "@medusajs/types"
+import { WorkflowData, createWorkflow } from "@medusajs/workflows-sdk"
+
+/*
+  - [] Create Tax Lines
+  - [] Authorize Payment
+    - fail:
+      - [] Delete Tax lines
+  - [] Reserve Item from inventory (if enabled)
```

Comment:
Should we reserve the items first, since that doesn't require interacting with an external service, and it serves as some sort of a lock, avoiding the possibility of us authorizing a payment for an item that is no longer available.